Sources/Models/CalculationParameters.swift

```diff
@@ -79,3 +82,253 @@ public struct CalculationParameters: Codable, Equatable {
         }
     }
 }
+
+public extension CalculationParameters {
+    /// Returns the recommended calculation parameters for the specified country code.
+    static func recommended(forCountryCode code: String) -> CalculationParameters? {
+        mapped.first { $0.value.contains(code) }?.key.params
+    }
+
+    #if canImport(CoreLocation)
+    /// Returns the recommended calculation parameters  for the specified geographic data.
+    static func recommended(for placemark: CLPlacemark) -> CalculationParameters? {
+        guard let countryCode = placemark.isoCountryCode else { return nil }
+        return recommended(forCountryCode: countryCode)
+    }
+    #endif
+
+    private static let mapped: [CalculationMethod: [String]] = [
+        .egyptian: ["EG", "LY", "SD"],
+        .karachi: ["AF", "IN", "PK", "TN"],
+        .ummAlQura: ["BH", "IQ", "OM", "SA", "SY", "YE"],
+        .dubai: ["AE"],
+        .moonsightingCommittee: ["CA", "GB"],
+        .northAmerica: ["NO", "US"],
+        .kuwait: ["KW"],
+        .qatar: ["QA"],
+        .singapore: ["SG"],
+        .tehran: ["IR"],
+        .turkey: ["TR"],
+        .muslimWorldLeague: [
```

Comment:
I think we can probably just leave this one out and make it the default value in the `recommended(forCountryCode:)` method so that it doesn't need to be optional.